ios/FluentUI/People Picker/AvatarViewVnext.swift

```diff
@@ -0,0 +1,336 @@
+        case .offline:
+            imageName = "ic_fluent_presence_offline_16_regular"
+        case .unknown:
+            imageName = "ic_fluent_presence_unknown_16_regular"
+        case .blocked:
+            imageName = "ic_fluent_presence_blocked_16_regular"
+        }
+
+        return Image(imageName,
+                     bundle: FluentUIFramework.resourceBundle)
+    }
+}
+
+@objc(MSFAvatarVnextState)
+/// Properties available to customize the state of the avatar
+public class AvatarVnextState: NSObject, ObservableObject {
+    @objc @Published public var image: UIImage?
+    @objc @Published public var primaryText: String?
+    @objc @Published public var secondaryText: String?
+    @objc @Published public var ringColor: UIColor?
+    @objc @Published public var backgroundColor: UIColor?
+    @objc @Published public var foregroundColor: UIColor?
+    @objc @Published public var presence: AvatarVnextPresence = .none
+}
+
+/// Representation of design tokens to buttons at runtime which interfaces with the Design Token System auto-generated code.
+/// Updating these properties causes the SwiftUI button to update its view automatically.
+public class AvatarTokens: ObservableObject {
+    @Published public var avatarSize: CGFloat!
+    @Published public var borderRadius: CGFloat!
+    @Published public var textFont: UIFont!
+
+    @Published public var ringDefaultColor: UIColor!
+    @Published public var ringGapColor: UIColor!
+    @Published public var ringThickness: CGFloat!
+    @Published public var ringInnerGap: CGFloat!
+    @Published public var ringOuterGap: CGFloat!
+
+    @Published public var presenceIconSize: CGFloat!
+    @Published public var presenceIconOutlineThickness: CGFloat!
+    @Published public var presenceOutlineColor: UIColor!
+
+    @Published public var backgroundDefaultColor: UIColor!
+    @Published public var foregroundDefaultColor: UIColor!
+
+    public var style: AvatarVnextStyle!
+    public var size: AvatarVnextSize!
+
+    public init(style: AvatarVnextStyle,
+                size: AvatarVnextSize) {
+        self.style = style
+        self.size = size
+        self.themeAware = true
+
+        didChangeAppearanceProxy()
+    }
+
+    @objc open func didChangeAppearanceProxy() {
+        var appearanceProxy: ApperanceProxyType
+
+        switch style {
+        case .default, .none:
+            appearanceProxy = StylesheetManager.S.AvatarTokens
+        case .noring:
+            appearanceProxy = StylesheetManager.S.NoRingAvatarTokens
+        case .unauthenticated:
+            appearanceProxy = StylesheetManager.S.UnauthenticatedAvatarTokens
+        case .overflow:
+            appearanceProxy = StylesheetManager.S.OverflowAvatarTokens
+        case .group:
+            appearanceProxy = StylesheetManager.S.GroupAvatarTokens
+        }
+
+        ringDefaultColor = appearanceProxy.ringDefaultColor
+        ringGapColor = appearanceProxy.ringGapColor
```

Comment:
I noticed that the ring gap color does not follow the background color in the alternate background toggle, was this intentional?

Reply:
On the figma file it's specified that both inner and outer rings should be Neutral Background. So I believe this is intentional. Started a thread with design to confirm anyways.
Thanks!

Reply:
@sophialee0416 I added code to support the "transparent" background of the presence icons and the outer ring gap (the inner gap is supposed to be the ringGapColor).
Please check the screenshots.